Lib/_pylong.py

```diff
@@ -82,7 +86,26 @@ def inner(n, w):
 
 def int_to_decimal_string(n):
     """Asymptotically fast conversion of an 'int' to a decimal string."""
-    return str(int_to_decimal(n))
+    w = n.bit_length()
+    if w > 900_000 and _decimal is not None:
+        return str(int_to_decimal(n))
+
+    def inner(n, w):
+        if w <= 1000:
+            return str(n)
+        w2 = w >> 1
+        d = pow10_cache.get(w2)
+        if d is None:
+            d = pow10_cache[w2] = 10**w2
```

Comment:
```suggestion
            d = pow10_cache[w2] = 5**w2 << w2 # 10**i = (5*2)**i = 5**i * 2**i
```
Shifting is cheap, allowing the costly power to produce `w2` fewer bits, which are all necessarily 0.

Reply:
Isn't it what the power implementation in C does? I do not see any measurable difference.

Reply:
I do see a measurable difference on my box, a few percent on large values. It's also why the existing `_str_to_int_inner()` uses `w5pow()` instead of using a `w10pow()`. Bjorn Martinsson found it faster on his box at the time too, when he wrote that function. Many people were interested in `_pylong` at the start, and there's no trick in the code that wasn't widely tried and debated at length then.

CPython's `i**j` does no special-casing of trailing 0 bits in `i`.  You can easily confirm that by, e.g., typing `x = 1024 ** 100000000` in a Python shell and noting that it takes significant time to complete. An equivalent shift appears instantaneous.